### ISSUE TITLE
Prediction values for a classifier trained on all-zeros target

### ISSUE THREAD

**Issue by sebov** [onnx/sklearn-onnx#1007]
Hi,
I noticed something unexpected when handling a classifier trained on all-zeros target `y`. In such the case, the original classifier's predictions on the training `x` are as expected (equal to `y`). However, the created ONNX counterpart classifier returns predictions in a form of all-ones. 

On the other hand, prediction proba responses for the original and the ONNX classifiers' are both as expected.

Is this a feature or a bug?

Please find a minimal example below:

```python
import numpy as np
import onnxruntime as rt
from skl2onnx import to_onnx
from sklearn.tree import DecisionTreeClassifier

x = np.eye(5, dtype=int)
y = np.array([0, 0, 0, 0, 0])

cl = DecisionTreeClassifier()
cl = cl.fit(x, y)

print("------------- sklearn DecisionTreeClassifier")
print(f"{cl.predict(x)=}")
print(f"{cl.predict_proba(x)=}")
print(f"{cl.classes_=}")

onx = to_onnx(cl, x)
sess = rt.InferenceSession(onx.SerializeToString())
res = sess.run(None, {"X": x})
print()
print("------------- ONNX")
print(res)
```

The result:
```
------------- sklearn DecisionTreeClassifier
cl.predict(x)=array([0, 0, 0, 0, 0])
cl.predict_proba(x)=array([[1.],
       [1.],
       [1.],
       [1.],
       [1.]])
cl.classes_=array([0])

------------- ONNX
[array([1, 1, 1, 1, 1], dtype=int64), [{0: 1.0}, {0: 1.0}, {0: 1.0}, {0: 1.0}, {0: 1.0}]]
```

```
Package              Version
-------------------- -------
coloredlogs          15.0.1
flatbuffers          23.5.26
humanfriendly        10.0
joblib               1.3.1
mpmath               1.3.0
numpy                1.25.1
onnx                 1.14.0
onnxconverter-common 1.13.0
onnxruntime          1.15.1
packaging            23.1
pip                  22.0.4
protobuf             4.23.4
scikit-learn         1.2.2
scipy                1.11.1
setuptools           58.1.0
skl2onnx             1.14.1
sympy                1.12
threadpoolctl        3.2.0
typing_extensions    4.7.1
```

**Comment by sebov** [comment]
Actually, I checked also other possible constant targets `y` and noticed similar behavior, e.g.:

```python
y = np.array([2, 2, 2, 2, 2])

y = np.array([-1, -1, -1, -1, -1])

...
```

Also give the ONNX classifier prediction response of the form 

```
array([1, 1, 1, 1, 1], dtype=int64)
```

**Comment by xadupre** [comment]
Can you try with `to_onnx(cl, x, options={"zipmap": False})`. The output is then a tensor and not a list of dictionaries.

**Comment by xadupre** [comment]
This happens when all the expected labels are the same. This case is not tested since it is not common but I guess it may happen when the dataset is split in such a way that one piece contain only one label. We should probably try to address that case for all the classifiers.
